sdk/src/v2_api/reader.rs

```diff
@@ -0,0 +1,267 @@
+
+    /// Create a manifest store Reader from existing c2pa_data and a stream
+    /// You can use this to validate a remote manifest or a sidecar manifest
+    /// # Arguments
+    /// * `c2pa_data` - The c2pa data (a manifest store in JUMBF format)
+    /// * `format` - The format of the stream
+    /// * `stream` - The stream to verify the store against
+    /// # Returns
+    /// A reader for the manifest store
+    /// # Errors
+    /// If the c2pa_data is not valid, or severe errors occur in validation
+    /// validation status should be checked for non severe errors
+    #[async_generic(async_signature(
+        c2pa_data: &[u8],
+        format: &str,
+        mut stream: impl Read + Seek + Send,
+    ))]
+    pub fn from_manifest_data_and_stream(
+        c2pa_data: &[u8],
+        format: &str,
+        mut stream: impl Read + Seek + Send,
+    ) -> Result<Reader> {
+        let mut validation_log = DetailedStatusTracker::new();
+
+        // first we convert the JUMBF into a usable store
+        let store = Store::from_jumbf(c2pa_data, &mut validation_log)?;
+
+        if _sync {
+            Store::verify_store(
+                &store,
+                &mut ClaimAssetData::Stream(&mut stream, format),
+                &mut validation_log,
+            )?;
+        } else {
+            Store::verify_store_async(
+                &store,
+                &mut ClaimAssetData::Stream(&mut stream, format),
+                &mut validation_log,
+            )
+            .await?;
+        }
+
+        Ok(Reader {
+            manifest_store: ManifestStore::from_store(&store, &validation_log),
+        })
+    }
+
+    /// Get the manifest store as a JSON string
+    pub fn json(&self) -> String {
+        self.manifest_store.to_string()
+    }
+
+    /// Get the validation status of the manifest store
+    /// # Returns
+    /// The validation status of the manifest store
+    /// # Example
+    /// ```no_run
+    /// use c2pa::Reader;
+    /// let stream = std::io::Cursor::new(include_bytes!("../../tests/fixtures/CA.jpg"));
+    /// let reader = Reader::from_stream("image/jpeg", stream).unwrap();
+    /// let status = reader.validation_status();
+    /// ```
+    /// # Note
+    /// The validation status should be checked for validation errors`
+    pub fn validation_status(&self) -> Option<&[ValidationStatus]> {
+        self.manifest_store.validation_status()
+    }
+
+    /// Get the active manifest
+    /// # Returns
+    /// The active manifest if it exists
+    pub fn active(&self) -> Option<&Manifest> {
+        self.manifest_store.get_active()
+    }
+
+    /// Get the active manifest label
+    pub fn active_label(&self) -> Option<&str> {
+        self.manifest_store.active_label()
+    }
+
+    /// Return a Manifest for a given label
+    /// # Arguments
+    /// * `label` - The label of the manifest to return
+    /// # Returns
+    /// The manifest if it exists
+    pub fn get(&self, label: &str) -> Option<&Manifest> {
```

Comment:
Since this is on an `impl` called `Reader` should this be `get_manifest`, `manifest` (maybe more Rusty), or `manifest_by_label`?

From the [_Rust API Guidelines_](https://rust-lang.github.io/api-guidelines/naming.html#c-getter):
>The `get` naming is used only when there is a single and obvious thing that could reasonably be gotten by a getter. For example [`Cell::get`](https://doc.rust-lang.org/std/cell/struct.Cell.html#method.get) accesses the content of a `Cell`.

I'm not sure if it's totally obvious that calling `get` on a `Reader` returns a `Manifest` (could it be a Manifest Store?), but if it is, I'm ok with leaving as is.

Reply:
Well it used to be called ManifestStore, and I suppose it made more sense that way. But the Reader is a ManifestStore reader.